### DISCUSSION TITLE
[Version 2] Public Logbook Landing Pages

### DISCUSSION THREAD

**Discussion by magicbug** [magicbug/Cloudlog#1226] [General]
With the addition of Logbooks and Cloudlog heading towards handling more than 1 user with separate logs and other features it will mean that the dashboard will need to be hidden first behind a "login" I know this is a change and will annoy some people but progress and all that.

What I'm proposing is that we create public landing pages for logbooks, each logbook will be issued a UUID that can be called and then a public landing page will be displayed similar to the Dashboard but with improvements, at first this needs to be MVP just to get it "live" therefore I'm going to propose the following

- Map showing last weeks worth of QSOs
- last 15QSOs or perhaps a JS driven table that can do back a few weeks
- Radio status (option to turn it off)
- QSO Stats (can be turned off)
- Search function but "ONLY" for that logbook group of locations
- Gridsquare map (useful for sat ops when trying to chase squares from others!)

Could be possible to add a few flow text area for announcements if you're going a rove, portable, sota type thing.

I'm thinking this would be https://cloudlog.domain/index.php/public/uuid

Looking for feedback before this development takes place but its going to be in some shape or form.

**Comment by tonyc** [comment]
I like the general idea, although I'm not too keen on the UUID for the public URL. If the idea is to share a public logbook, it might make it difficult to quickly share. Would it be possible to generate a slug based off a primary callsign, or something similar? e.g. `index.php/public/callsign`. 

Or if that's not possible, consider not using a full UUID, but something shorter, yet still random? Is the UUID essentially the PK of the account/log in the database?

**Reply by magicbug** [reply]
I was going to generate a random ID string just for access totally separate to the main PK of the database. I cant really use the callsign as you might use that twice? so that's why I thought this route..

**Reply by AndreasK79** [reply]
Why not do something similar to what @tonyc suggests, but use callsign + a number, but only if the slug is taken? That will be much easier to remember, than a unique string.

**Comment by M0TZO** [comment]
Sounds like a good plan to me.

**Comment by dg9vh** [comment]
Would also say, this is a good idea.

Paul GIBSON ***@***.***> schrieb am Mi., 13. Okt. 2021, 23:43:

**Comment by timkirby** [comment]
This sounds great. I like the idea of being able to make some sort of announcement too, as ou suggest

Although I know iFrames are a no-no these days, I'd like to have the ability to embed a landing page, or a list of last ten QSOs or whatever on QRZ or similar. I get around it now by using the Clublog snippet on QRZ, but it would be quite nice to be able to feed from Cloudlog directly.

**Reply by M0TZO** [reply]
Tim, `https://<your-url>/index.php/widgets/qsos`

**Comment by timkirby** [comment]
Ha, if I knew that, I’d forgotten!!

Thanks Paul :-)

**Comment by timkirby** [comment]
Mmm, except that it doesn’t work with QRZ.COM <http://qrz.com/> pages which is the most likely place for it to be used for most people.

That’s completely an issue at QRZ.COM <http://qrz.com/> rather than with Cloudlog. I’ve a feeling they block everything except some known domains. Maybe I will drop Michael (Clublog) an email and see how he got his code accepted at QRZ. He may be able to point us in the right direction. I’ve a feeling I have asked QRZ about something like this before and been turned down.

**Reply by simonszu** [reply]
@timkirby Do you have any further information about that? Currently i try to display my Cloudlog Widget on QRZ, but all i get is an error that the connection was denied. 
However, i can undertand QRZ not allowing this, since they charge money for their XML API, and this widget would be a way how someone can bypass this API and therefore QRZ not getting money from the specific OP.

**Reply by magicbug** [reply]
Not sure how this gets around paying QRZ money for API its just a widget to show QSOs worked, but as their a commercial site they can do as they choose.

**Comment by 112bzo** [comment]
What I personally will like is a twitter API integration

I now tweet my made contacts via HRDlog, but notice that it regularly crashes and doesn't send anything.

**Reply by magicbug** [reply]
Not going to happen, its utterly annoying :)

**Comment by magicbug** [comment]
Things are progressing nicely with the V2 code in the station logbooks branch, this item is now moving close to the top of the todo.

**Comment by M0TZO** [comment]
After sorting out a Cloudlog for another club/group, I wondered if would be possible to have a file/document area for licences, back up logs, maybe photos etc?

**Reply by magicbug** [reply]
I'd suggest creating a new thread to discuss a new feature.

**Comment by timkirby** [comment]
Hi Simon,

I have mine working on both GW4VXE and GW4MM QRZ pages, so I don’t think they are blocking anything now.

The code I am using is

<p><em><em><span style="font-size:18px"><iframe height="300" src="https://g4vxe.cloudlog.co.uk/index.php/widgets/qsos/GW4MM" width="700"></iframe></span></em></em></p>

GW4MM is the name of the station logbook

Hope that helps,

Tim

**Comment by timkirby** [comment]
Paul M0TZO kindly reminded me that, in this case, the GW4MM on the end of the URL is the ‘Public Slug’, which you define on the Station Logbook screen.

Tim

**Comment by R1BLH** [comment]
Looks great!
Is it possible to hide QSO time from widgets on the public landing page?
